### PR TITLE
Reserve vectors before pushing and other code quality changes

### PR DESCRIPTION
**Pull request by sfan5**

### CODE REVIEW

File: src/client/clouds.cpp [hunk 1]
```diff
@@ -170,7 +170,7 @@ void Clouds::render()
 
 	// Read noise
 
-	std::vector<char> grid(m_cloud_radius_i * 2 * m_cloud_radius_i * 2); // vector<bool> is broken
+	std::vector<bool> grid(m_cloud_radius_i * 2 * m_cloud_radius_i * 2);
```
Comment (SmallJoker) [inline src/client/clouds.cpp:173]: https://github.com/minetest/minetest/pull/10660#pullrequestreview-536104315

File: src/client/hud.cpp [hunk 2]
```diff
@@ -522,8 +522,8 @@ void Hud::drawLuaElements(const v3s16 &camera_offset)
 				client->getMinimap()->drawMinimap(rect);
 				break; }
 			default:
-				infostream << "Hud::drawLuaElements: ignoring drawform " << e->type <<
-					" of hud element ID " << i << " due to unrecognized type" << std::endl;
+				infostream << "Hud::drawLuaElements: ignoring drawform " << e->type
```
Comment (SmallJoker) [inline src/client/hud.cpp:525]: The server HUD ID might still be helpful here to figure out which HUD is affected.

File: src/client/clouds.cpp (hunk 1 above)
Comment (sfan5) [reply src/client/clouds.cpp:173]: Yeah I remember but I couldn't find a reason why `vector<bool>` shouldn't work here and it in fact does.

File: src/client/hud.cpp (hunk 2 above)
Comment (sfan5) [reply src/client/hud.cpp:525]: The small reorganization up top unfortunately makes this impossible.

**Review by SmallJoker — APPROVED** [review]
LGTM

**Pull request merged**